gems/aws-sdk-core/lib/aws-sdk-core/plugins/jsonvalue_converter.rb

```diff
@@ -11,15 +11,31 @@ class Handler < Seahorse::Client::Handler
 
         def call(context)
           context.operation.input.shape.members.each do |m, ref|
-            if ref['jsonvalue']
-              param_value = context.params[m]
-              unless param_value.respond_to?(:to_json)
-                raise ArgumentError, "The value of params[#{m}] is not JSON serializable."
+            convert_jsonvalue(m, ref, context.params)
+          end
+          @handler.call(context)
+        end
+
+        def convert_jsonvalue(m, ref, params)
+          return if params.nil? || params[m].nil?
+
+          if ref['jsonvalue']
+            param_value = params[m]
+            unless param_value.respond_to?(:to_json)
+              raise ArgumentError, "The value of params[#{m}] is not JSON serializable."
+            end
+            params[m] = param_value.to_json
+          elsif ref.shape.is_a?(Seahorse::Model::Shapes::StructureShape)
```

Comment:
I think we need to handle other shape types, like Map and Union?

Reply:
Better yet, is there a solution that doesn't involve shape checks here?

Reply:
The only other option (which is what I started with) was to check `respond_to?(:members)` and `respond_to?(:member)` and `respond_to?(:value)`.  But I think thats worse than just being shape based.

Reply:
Yeah, agreed, then shape is fine, assuming we cover them all.